src/components/radio-button/radio-button.pw.tsx

```diff
@@ -0,0 +1,815 @@
+import { expect, test } from "@playwright/experimental-ct-react17";
+import React from "react";
+import {
+  fieldHelpPreview,
+  getComponent,
+  tooltipPreview,
+} from "../../../playwright/components/index";
+import {
+  radiobutton,
+  radiobuttonComponent,
+  radiobuttonGroup,
+  radiobuttonGroupIcon,
+  radiobuttonGroupLegend,
+  radiobuttonIcon,
+  radiobuttonInlineFieldHelp,
+  radiobuttonLabel,
+  radiobuttonRole,
+  radiobuttonSvg,
+} from "../../../playwright/components/radio-button";
+import {
+  CHARACTERS,
+  COLOR,
+  SIZE,
+  VALIDATION,
+} from "../../../playwright/support/constants";
+import {
+  assertCssValueIsApproximately,
+  checkAccessibility,
+  verifyRequiredAsteriskForLegend,
+} from "../../../playwright/support/helper";
+import { RadioButtonGroupProps } from "../../../src/components/radio-button/radio-button-group.component";
+import RadioButton, {
+  RadioButtonProps,
+} from "../../../src/components/radio-button/radio-button.component";
+import {
+  RadioButtonComponent,
+  RadioButtonGroupComponent,
+  Default,
+  DifferentLabelSpacing,
+  DisableRadioButtons,
+  EnableAdaptiveBehaviour,
+  InlineRadioButtons,
+  ReverseRadioButtons,
+  WithCustomStyledLabels,
+  WithFieldHelp,
+  WithInlineLegend,
+  WithLargeRadioButtons,
+  WithLeftMargin,
+  WithLegendAndLabels,
+} from "./components.test-pw";
+
+const testData = [CHARACTERS.DIACRITICS, CHARACTERS.SPECIALCHARACTERS];
+const radioContainerWidth = 400;
+const radioInputWidth = 16;
+const labelContainerWidth = 40;
+
+test.describe("should render RadioButton component", () => {
+  test("should render RadioButton component with data-component", async ({
```

Comment:
Not a blocking issue, but following on from Ed's suggestion, we don't need to mention the component name in each test title if it's already in the describe. Something like `"should check data-component is ${CHARACTERS.STANDARD}"` should cover it.